lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2848,6 +2851,30 @@ SubExprInfo ExprEmitter::visitTypeOp(EnumConstantOp op) {
   return {Selection, IsUnsigned};
 }
 
+SubExprInfo ExprEmitter::visitTypeOp(UnionExtractOp op) {
+  if (hasSVAttributes(op))
+    emitError(op, "SV attributes emission is unimplemented for the op");
+  emitSubExpr(op.getInput(), Selection);
+
+  // Check if this union type has been padded.
```

Comment:
This could get expensive to compute. We might think about caching the result at some point, but that seems like pre-optimization to do now.